py/torch_tensorrt/dynamo/conversion/impl/split.py

```diff
@@ -0,0 +1,81 @@
+from typing import Any, Dict, List, Optional, Sequence, Tuple, Union, cast
+
+import numpy as np
+import torch
+import torch_tensorrt as trt
+from torch import Tensor
+from torch.fx.node import Target
+from torch_tensorrt.dynamo._SourceIR import SourceIR
+from torch_tensorrt.dynamo.conversion.impl.shape import get_shape_with_dynamic_shape
+from torch_tensorrt.fx.converters.converter_utils import (
+    has_dynamic_shape,
+    set_layer_name,
+)
+from torch_tensorrt.fx.types import TRTNetwork, TRTTensor
+
+
+def split(
+    network: TRTNetwork,
+    target: Target,
+    source_ir: Optional[SourceIR],
+    name: str,
+    input: TRTTensor,
+    split_size_or_sections: Union[int, List[int]],
+    dim: int = 0,
+) -> Union[TRTTensor, Sequence[TRTTensor]]:
+    if not isinstance(input, TRTTensor):
+        raise RuntimeError(
+            f"split received input {input} that is not part " "of the TensorRT region!"
+        )
+
+    dynamic_shape = has_dynamic_shape(input.shape)
+    if dynamic_shape > 0:
+        # Check whether slice target dim is dynamic shape dim
+        assert input.shape[dim] != -1, "Can't chunk on dynamic shape dimension!"
+
+    split_sizes = []
+    if isinstance(split_size_or_sections, int):
+        split_sizes.append(split_size_or_sections)
+    else:
+        for split_size_or_section in split_size_or_sections:
+            split_sizes.append(cast(int, split_size_or_section))
```

Comment:
@gs-olive Just kept this `cast`. Is it possible that the individual element is` split_size_or_section` is not type casted? In case yes, I will keep this, else I will remove.

Reply:
It should be okay to keep this one, but I don't think the `cast` operator actually performs the cast, it instead just informs the linter of the type of that object.

Reply:
Ok will remove. So when you say you checked `mypy` linting, did you run it on the terminal or you could see in the below linting fail? Because I do not explicitly see the failure in the linting test failure below